### PR TITLE
BENCH-191 BENCH-192 BENCH-193 BENCH-197

### PR DESCRIPTION
**Pull request by milkshakeuk**
* Hide Payments endpoints
* Use Inventory tag for Products and Categories endpoints
* Add descriptions to openApi document tags
* Update request model names to remove Dto suffix ([BENCH-191](https://answerconsulting.jira.com/browse/BENCH-191))
* Remove now unused schemas from openApi ([BENCH-197?](https://answerconsulting.jira.com/browse/BENCH-197))
* Consolodated ValidationProblemDetails into ProblemDetails for openApi ([BENCH-192](https://answerconsulting.jira.com/browse/BENCH-192))
* Remove Answer.King prefix ([BENCH-193](https://answerconsulting.jira.com/browse/BENCH-193))
* Add custom JsonConverter to hide simplify ProductId and CategoryId when de/serialising
* Simplified storage of ProductId and CategoryId in LiteDb.
* Use camelcase for fieldNames in LiteDb

image

### CODE REVIEW

File: src/Answer.King.Api/Controllers/PaymentsController.cs [hunk 1]
```diff
@@ -68,9 +68,9 @@ public async Task<IActionResult> GetOne(long id)
     /// <response code="400">When invalid parameters are provided.</response>
     // POST api/payments
     [HttpPost]
-    [ProducesResponseType(typeof(Payment), StatusCodes.Status201Created)]
+    [ProducesResponseType(typeof(Domain.Repositories.Models.Payment), StatusCodes.Status201Created)]
     [ProducesResponseType(typeof(ValidationProblemDetails), StatusCodes.Status400BadRequest)]
```
Comment (kirbyAlasdairAnswer) [inline src/Answer.King.Api/Controllers/PaymentsController.cs]: Should this be ProblemDetails?

File: src/Answer.King.Api/OpenApi/ValidationProblemDetailsSchemaFilter.cs [hunk 2]
```diff
@@ -10,7 +10,7 @@ public class ValidationProblemDetailsSchemaFilter : ISchemaFilter
 {
     public void Apply(OpenApiSchema schema, SchemaFilterContext context)
     {
-        if (context.Type != typeof(Microsoft.AspNetCore.Mvc.ValidationProblemDetails))
+        if (context.Type != typeof(Microsoft.AspNetCore.Mvc.ProblemDetails))
```
Comment (kirbyAlasdairAnswer) [inline src/Answer.King.Api/OpenApi/ValidationProblemDetailsSchemaFilter.cs:13]: Is this mapping any remaining ValidationProblemDetails object to ProblemDetails? Will they appear as ProblemDetails in the openApi Schema?

File: src/Answer.King.Api/Controllers/PaymentsController.cs (hunk 1 above)
Comment (milkshakeuk) [reply src/Answer.King.Api/Controllers/PaymentsController.cs:72]: No, don't worry when it generates it's just `ProblemDetails`.

Comment (milkshakeuk) [reply src/Answer.King.Api/Controllers/PaymentsController.cs:72]: No, don't worry when it generates it's just `ProblemDetails`.probably yes, payments api is hidden anyway.

File: src/Answer.King.Api/OpenApi/ValidationProblemDetailsSchemaFilter.cs (hunk 2 above)
Comment (milkshakeuk) [reply src/Answer.King.Api/OpenApi/ValidationProblemDetailsSchemaFilter.cs:13]: Yes 😊

File: src/Answer.King.Api/Controllers/PaymentsController.cs (hunk 1 above)
Comment (milkshakeuk) [reply src/Answer.King.Api/Controllers/PaymentsController.cs]: Yes, updated, this is hidden from openApi anyway

**Review by kirbyAlasdairAnswer — APPROVED** [review]

**Review by mbaltrimas — APPROVED** [review]

**Pull request merged**